retype.py

```diff
@@ -138,7 +139,8 @@ def retype_file(src, pyi_dir, targets, *, quiet=False, hg=False):
 
     Type comments in sources are normalized to type annotations.
     """
-    with open(src) as src_file:
+    src_encoding = tokenize.open(src).encoding
```

Comment:
This is slower than necessary and generates a ResourceWarning. Could you use `tokenize.detect_encoding()` instead on the file open below? Then you only need to `src_file.seek(0, 0)` to reset the file.